### PR TITLE
Remove dump cluster from gen_cluster

### PR DESCRIPTION
**Pull request by fjetter**
I don't see a lot of value for having this still around. However, it is spamming the working directory and makes test execution (if failed) much slower.

### CODE REVIEW

**Comment by hendrikmakait** [comment]
CI isn't happy with this PR, need help fixing this up?

File: distributed/utils_test.py [hunk 1]
```diff
@@ -1122,41 +1106,6 @@ async def async_fn_outer():
     return _
 
 
-async def dump_cluster_state(
```
Comment (jrbourbeau) [inline distributed/utils_test.py:1125]: Another option is to leave this functionality around but still disable it in `gen_cluster`. I'm not sure how much value there is to having `dump_cluster_state` around for diagnosing tricky deadlock situations.

**Review by jrbourbeau — APPROVED** [review]
Thanks @fjetter. I left one non-blocking comment but otherwise this LGTM

File: distributed/utils_test.py (hunk 1 above)
Comment (fjetter) [reply distributed/utils_test.py:1125]: The functionality itself is part of Client. I'm not entirely sure what this utility function offers but I'd rather not maintain it.

**Pull request merged**